nion/swift/model/Graphics.py

```diff
@@ -1706,7 +1749,16 @@ def draw(self, ctx: DrawingContextLike, ui_settings: UISettings.UISettings, mapp
     def label_position(self, mapping: CoordinateMappingLike, font_metrics: UISettings.FontMetrics, padding: float) -> typing.Optional[Geometry.FloatPoint]:
         p = Geometry.FloatPoint.make(mapping.map_point_image_norm_to_widget(self.position))
         return p + Geometry.FloatPoint(-self.cross_hair_size - font_metrics.height * 0.5 - padding * 2, 0.0)
+    
+    @property
+    def used_role(self) -> typing.Optional[str]:
+        return "fourier_mask"
 
+    def get_mask(self, data_shape: DataAndMetadata.ShapeType, calibrated_origin: typing.Optional[Geometry.FloatPoint] = None) -> DataAndMetadata._ImageDataType:
+        mask = numpy.zeros(data_shape)
+        x, y = self.position
+        mask[int(x*data_shape[0]), int(y*data_shape[1])] = 1
```

Comment:
There should be a check to ensure `x` and `y` are in the range [0,1]. If either are out of range, no value should be set.

Reply:
I have implemented the check. points outside the range will not create the mask.